datafusion/core/src/execution/context/mod.rs

```diff
@@ -687,7 +688,31 @@ impl SessionContext {
             LogicalPlan::Statement(Statement::SetVariable(stmt)) => {
                 self.set_variable(stmt).await
             }
-
+            LogicalPlan::Prepare(Prepare {
```

Comment:
I am worried that `LogicalPlan::Prepare` seems to be runnable even if we run `sql_with_options` and disable statements. 

https://github.com/apache/datafusion/blob/aad77446f54c89cdaaa4f61909a3e0230b8cc326/datafusion/core/src/execution/context/mod.rs#L605-L614

I realize that this PR does not change if Prepare is a statement or not, but it is the first that actually makes `Prepare` do something


Could you please add a test like this (but for `PREPARE` statements) and make sure the statement can't be executed if statements are disabled?

https://github.com/apache/datafusion/blob/6692382f22f04542534bba0183cf0682fd932da1/datafusion/core/tests/sql/sql_api.rs#L99-L114


Perhaps (as a follow on PR) we can make `LogicalPlan::Prepare` a statement https://docs.rs/datafusion/latest/datafusion/logical_expr/enum.Statement.html

Reply:
Temporarily added PREPARE to BadPlanVisitor and added a test for this in https://github.com/apache/datafusion/pull/13242/commits/472976639ac5a7ebe6914205f4026c3ee440d099
I will make LogicalPlan::Prepare a statement in a follow-up PR.